src/Router/ModelParameterResolver.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace eLife\Journal\Router;
+
+use Cocur\Slugify\SlugifyInterface;
+use eLife\ApiSdk\Model;
+
+final class ModelParameterResolver implements ParameterResolver
+{
+    private $slugify;
+
+    public function __construct(SlugifyInterface $slugify)
+    {
+        $this->slugify = $slugify;
+    }
+
+    public function resolve(string $route, array $parameters) : array
+    {
+        if (!isset($parameters[0])) {
+            return $parameters;
+        }
+
+        $model = $parameters[0];
+        unset($parameters[0]);
+        $new = [];
+
+        if ($model instanceof Model\ArticleVersion) {
+            $new = ['id' => $model->getId()];
+        } elseif ($model instanceof Model\BlogArticle) {
+            $new = ['id' => $model->getId(), 'slug' => $this->slugify->slugify($model->getTitle())];
+        } elseif ($model instanceof Model\Collection) {
+            $new = ['id' => $model->getId(), 'slug' => $this->slugify->slugify($model->getTitle())];
+        } elseif ($model instanceof Model\Event) {
+            $new = ['id' => $model->getId(), 'slug' => $this->slugify->slugify($model->getTitle())];
+        } elseif ($model instanceof Model\Interview) {
+            $new = ['id' => $model->getId(), 'slug' => $this->slugify->slugify($model->getInterviewee()->getPerson()->getPreferredName())];
+        } elseif ($model instanceof Model\LabsPost) {
+            $new = ['id' => $model->getId(), 'slug' => $this->slugify->slugify($model->getTitle())];
+        } elseif ($model instanceof Model\PodcastEpisode) {
+            $new = ['number' => $model->getNumber()];
+        } elseif ($model instanceof Model\PodcastEpisodeChapterModel) {
+            $new = ['number' => $model->getEpisode()->getNumber(), '_fragment' => $model->getChapter()->getTime()];
+        } elseif ($model instanceof Model\PressPackage) {
+            $new = ['id' => $model->getId(), 'slug' => $this->slugify->slugify($model->getTitle())];
+        } elseif ($model instanceof Model\Subject) {
+            $new = ['id' => $model->getId()];
+        }
```

Comment:
Probably, it relies on there never being a route with a parameter `0`, but I can't imagine that'd happen.